packages/vue-app/template/router.js

```diff
@@ -3,16 +3,32 @@ import Router from 'vue-router'
 import { interopDefault } from './utils'
 
 <% function recursiveRoutes(routes, tab, components, indentCount) {
-  let res = ''
+  let res = '', resMap = ''
   const baseIndent = tab.repeat(indentCount)
   const firstIndent = '\n' + tab.repeat(indentCount + 1)
   const nextIndent = ',' + firstIndent
   routes.forEach((route, i) => {
-    route._name = '_' + hash(route.component)
-    components.push({ _name: route._name, component: route.component, name: route.name, chunkName: route.chunkName })
+    if(route.components) {
+      let _name = '_' + hash(route.components.default)
+      resMap += firstIndent + tab + 'default: ' + (splitChunks.pages ? _name : `() => ${_name}.default || ${_name}`)
+      Object.keys(route.components).map(function(k){
```

Comment:
@karakum I think `for (const k in route.components) {` work here just as well -- would you please change it? Makes for better readability. Similarly, avoid having such a long line like:

```js
      resMap += firstIndent + tab + 'default: ' + (splitChunks.pages ? _name : `() => ${_name}.default || ${_name}`)
```

That can be split across multiple lines:

```js
      if (splitChunks.pages) {
        resMap += firstIndent + tab + 'default: ' + _name
      } else {
        resMap += firstIndent + tab + 'default: ' + `() => ${_name}.default || ${_name}`
      }
```

Which, despite some repetition, helps keep code easier to understand and maintain.